src/api.rs

```diff
@@ -0,0 +1,323 @@
+        settings.import_overlays = self.vmf_import_overlays;
+        settings.import_props = self.vmf_import_props;
+        settings.import_other_entities = self.vmf_import_entities;
+        settings.import_skybox = self.vmf_import_sky;
+        settings.scale = self.vmf_scale;
+
+        settings.brushes = if self.vmf_import_brushes {
+            BrushSetting::Import(geometry_settings)
+        } else {
+            BrushSetting::Skip
+        };
+
+        let path: PathBuf = if from_game {
+            GamePathBuf::from(path).into()
+        } else {
+            StdPathBuf::from(path).into()
+        };
+
+        self.jobs.push(AssetImportJob::Vmf {
+            path,
+            config: settings,
+        });
+
+        Ok(())
+    }
+
+    fn add_mdl_job(&mut self, path: &str, from_game: bool) -> PyResult<()> {
+        let mut settings = MdlConfig::new(self.material_config);
+        settings.import_animations = self.mdl_import_animations;
+
+        let path = if from_game {
+            GamePathBuf::from(path).into()
+        } else {
+            StdPathBuf::from(path).into()
+        };
+
+        self.jobs.push(AssetImportJob::Mdl {
+            path,
+            config: settings,
+        });
+
+        Ok(())
+    }
+
+    fn add_vmt_job(&mut self, path: &str, from_game: bool) -> PyResult<()> {
+        let path = if from_game {
+            GamePathBuf::from(path).into()
+        } else {
+            StdPathBuf::from(path).into()
+        };
+
+        self.jobs.push(AssetImportJob::Vmt { path });
+
+        Ok(())
+    }
+
+    fn add_vtf_job(&mut self, path: &str, from_game: bool) -> PyResult<()> {
+        let path = if from_game {
+            GamePathBuf::from(path).into()
+        } else {
+            StdPathBuf::from(path).into()
+        };
+
+        self.jobs.push(AssetImportJob::Vtf { path });
+
+        Ok(())
+    }
+
+    fn execute_jobs(&mut self, py: Python) -> PyResult<()> {
+        if self.jobs.is_empty() {
+            return Ok(());
+        }
+
+        let executor = self.consume()?;
+        let start = Instant::now();
+        info!("executing {} import jobs in parallel...", self.jobs.len());
+
+        // Use single process_each call with unified config
+        let unified_config = UnifiedAssetConfig {
+            material_config: self.material_config,
+        };
+
+        let jobs: Vec<AssetImportJob> = self.jobs.drain(..).collect();
+        executor.process_each(unified_config, jobs, || self.process_assets(py));
+
+        info!("jobs executed in {:.2} s", start.elapsed().as_secs_f32());
+        Ok(())
+    }
+
+    #[getter]
+    fn job_count(&self) -> usize {
+        self.jobs.len()
+    }
+}
+
+impl PyApiImporter {
+    fn consume(&mut self) -> PyResult<Executor<BlenderAssetHandler>> {
+        self.executor
+            .take()
+            .ok_or_else(|| PyRuntimeError::new_err("Importer already consumed"))
+    }
+
+    fn process_assets(&self, py: Python) {
```

Comment:
This is probably the same function as in PyImporter? Could it be a function that takes callback_ref defined in PyImporter, and PyImporter and this one could use the same function then to prevent duplicate code?